build-conventions/build.gradle

```diff
@@ -17,88 +17,89 @@ buildscript {
+  // with the java-gradle-plugin
+  automatedPublishing = false
+  plugins {
+    internalLicenseheaders {
+      id = 'elasticsearch.internal-licenseheaders'
+      implementationClass = 'org.elasticsearch.gradle.internal.conventions.precommit.LicenseHeadersPrecommitPlugin'
     }
+    eclipse {
+      id = 'elasticsearch.eclipse'
+      implementationClass = 'org.elasticsearch.gradle.internal.conventions.EclipseConventionPlugin'
+    }
+    publish {
+      id = 'elasticsearch.publish'
+      implementationClass = 'org.elasticsearch.gradle.internal.conventions.PublishPlugin'
+    }
+    licensing {
+      id = 'elasticsearch.licensing'
+      implementationClass = 'org.elasticsearch.gradle.internal.conventions.LicensingPlugin'
+    }
+    buildTools {
+      id = 'elasticsearch.build-tools'
+      implementationClass = 'org.elasticsearch.gradle.internal.conventions.BuildToolsConventionsPlugin'
+    }
+    versions {
+      id = 'elasticsearch.versions'
+      implementationClass = 'org.elasticsearch.gradle.internal.conventions.VersionPropertiesPlugin'
+    }
+    formatting {
+      id = 'elasticsearch.formatting'
+      implementationClass = 'org.elasticsearch.gradle.internal.conventions.precommit.FormattingPrecommitPlugin'
+    }
+  }
 }
 
 repositories {
-    mavenCentral()
-    gradlePluginPortal()
+  mavenCentral()
+  gradlePluginPortal()
 }
 
 dependencies {
-    api buildLibs.maven.model
-    api buildLibs.shadow.plugin
-    api buildLibs.apache.rat
-    compileOnly buildLibs.checkstyle
-    constraints {
-      api("org.eclipse.platform:org.eclipse.osgi:3.18.300") {
-        because("Use the same version as we do in spotless gradle plugin at runtime")
-      }
-    }
-    api(buildLibs.spotless.plugin) {
-      exclude module: "groovy-xml"
+  api buildLibs.maven.model
+  api buildLibs.shadow.plugin
+  api buildLibs.apache.rat
+  api buildLibs.nmcp
+  compileOnly buildLibs.checkstyle
+  constraints {
+    api("org.eclipse.platform:org.eclipse.osgi:3.18.300") {
+      because("Use the same version as we do in spotless gradle plugin at runtime")
     }
+  }
+  api(buildLibs.spotless.plugin) {
+    exclude module: "groovy-xml"
+  }
 }
 
 project.getPlugins().withType(JavaBasePlugin.class) {
-    java.getModularity().getInferModulePath().set(false);
-    eclipse.getClasspath().getFile().whenMerged { classpath ->
-        /*
-         * give each source folder a unique corresponding output folder
-         * outside of the usual `build` folder. We can't put the build
-         * in the usual build folder because eclipse becomes *very* sad
-         * if we delete it. Which `gradlew clean` does all the time.
-         */
-        classpath.getEntries().findAll{ s -> s instanceof SourceFolder }.eachWithIndex { s, i ->
-            s.setOutput("out/eclipse" + i)
-        }
+  java.getModularity().getInferModulePath().set(false);
+  eclipse.getClasspath().getFile().whenMerged { classpath ->
```

Comment:
unrelated: we might want to remove all the leftover eclipse related configuration to our build. we do not support that anymore.